crates/hir_ty/src/diagnostics/decl_check.rs

```diff
@@ -312,37 +330,47 @@ impl<'a, 'b> DeclValidator<'a, 'b> {
     fn validate_struct(&mut self, db: &dyn HirDatabase, struct_id: StructId) {
         let data = db.struct_data(struct_id);
 
-        // 1. Check the structure name.
+        let non_camel_case_allowed =
+            self.allowed(db, struct_id.into(), allow::NON_CAMEL_CASE_TYPES);
+        let non_snake_case_allowed = self.allowed(db, struct_id.into(), allow::NON_SNAKE_CASE);
+
+        // Check the structure name.
         let struct_name = data.name.to_string();
         let struct_name_replacement = if let Some(new_name) = to_camel_case(&struct_name) {
             let replacement = Replacement {
                 current_name: data.name.clone(),
                 suggested_text: new_name,
                 expected_case: CaseType::UpperCamelCase,
             };
-            Some(replacement)
+            if !non_camel_case_allowed {
+                Some(replacement)
+            } else {
+                None
+            }
```

Comment:
```suggestion
            if non_camel_case_allowed {
                None
            } else {
                Some(replacement)
            }
```

positive conditions are generally easier to read.

Reply:
I personally prefer putting the more likely path first, but I agree that suggested convention is more common :)

Reply:
That's also reasonable, though I usually deal with that by

```
if unlikely {
    return edge_case
}
base_case
```